### PR TITLE
Topic/clang format master at 0e768557

### PR DESCRIPTION
**Pull request by hppritcha**
second pass at running clang-format script against ompi, opal, and oshmem folders.  Tested a Intel OPA, Cray Aries, and x86_64 with Mellanox interconnect (UCX 1.10.1).

### CODE REVIEW

**Comment by hjelmn** [comment]
Does oshmem break if the headers are reordered? If so that is a bug and will need to be fixed eventually. Includes should not rely on ordering.

File: ompi/attribute/attribute.h [hunk 1]
```diff
@@ -155,17 +136,17 @@ typedef union ompi_attribute_fortran_ptr_t ompi_attribute_fortran_ptr_t;
 
 struct ompi_attribute_keyval_t {
     opal_object_t super;
-    ompi_attribute_type_t attr_type; /**< One of COMM/WIN/DTYPE. This
-                                       will be used to cast the
-                                       copy/delete attribute functions
-                                       properly and error checking */
-    int attr_flag; /**< flag field: contains "OMPI_KEYVAL_PREDEFINED",
-                      "OMPI_KEYVAL_F77"  */
+    ompi_attribute_type_t attr_type;            /**< One of COMM/WIN/DTYPE. This
```
Comment (gpaulsen) [inline ompi/attribute/attribute.h:139]: why the extra spaces?  makes no sense.

File: ompi/attribute/attribute.h [hunk 2]
```diff
@@ -155,17 +136,17 @@ typedef union ompi_attribute_fortran_ptr_t ompi_attribute_fortran_ptr_t;
...
+                                                  copy/delete attribute functions
+                                                  properly and error checking */
+    int attr_flag;                              /**< flag field: contains "OMPI_KEYVAL_PREDEFINED",
+                                                   "OMPI_KEYVAL_F77"  */
     ompi_attribute_fn_ptr_union_t copy_attr_fn; /**< Copy function for the
                                              attribute */
     ompi_attribute_fn_ptr_union_t delete_attr_fn; /**< Delete function for the
                                                attribute */
-    ompi_attribute_fortran_ptr_t extra_state; /**< Extra state of the attribute */
+    ompi_attribute_fortran_ptr_t extra_state;     /**< Extra state of the attribute */
```
Comment (gpaulsen) [inline ompi/attribute/attribute.h:149]: Does it interpret `/**` as meaning, please put some space here?

File: ompi/attribute/attribute_predefined.c [hunk 3]
```diff
@@ -162,30 +161,28 @@ int ompi_attr_create_predefined(void)
...
-        OMPI_SUCCESS != (ret = free_win(MPI_WIN_CREATE_FLAVOR)) ||
-        OMPI_SUCCESS != (ret = free_win(MPI_WIN_MODEL))) {
+    if (OMPI_SUCCESS != (ret = free_comm(MPI_TAG_UB)) || OMPI_SUCCESS != (ret = free_comm(MPI_HOST))
+        || OMPI_SUCCESS != (ret = free_comm(MPI_IO))
+        || OMPI_SUCCESS != (ret = free_comm(MPI_WTIME_IS_GLOBAL))
+        || OMPI_SUCCESS != (ret = free_comm(MPI_APPNUM))
+        || OMPI_SUCCESS != (ret = free_comm(MPI_LASTUSEDCODE))
+        || OMPI_SUCCESS != (ret = free_comm(MPI_UNIVERSE_SIZE))
+        || OMPI_SUCCESS != (ret = free_comm(MPI_FT))
+        || /* not #if conditional on OPAL_ENABLE_FT_MPI for ABI */
```
Comment (gpaulsen) [inline ompi/attribute/attribute_predefined.c:175]: Can possibly remove these references to `not #if conditional on OPAL_ENABLE_FT_MPI` now that FT is IN.

File: ompi/attribute/attribute_predefined.c [hunk 4]
```diff
@@ -194,11 +191,12 @@ static int create_comm(int target_keyval, bool want_inherit)
     ompi_attribute_fn_ptr_union_t del;
 
     keyval = -1;
-    copy.attr_communicator_copy_fn = (MPI_Comm_internal_copy_attr_function*)
-        (want_inherit ? MPI_COMM_DUP_FN : MPI_COMM_NULL_COPY_FN);
+    copy.attr_communicator_copy_fn = (MPI_Comm_internal_copy_attr_function
```
Comment (gpaulsen) [inline ompi/attribute/attribute_predefined.c:194]: What happened here?

Comment (hjelmn) [reply ompi/attribute/attribute_predefined.c:194]: This we can fix by using proper function pointer typedefs. I have a PR up to address it.

File: ompi/attribute/attribute.h (hunk 2 above)
Comment (hjelmn) [reply ompi/attribute/attribute.h:149]: It is aligning the comments. We really should avoid the /**<**/ doxygen comments and just put the comment above the strict member.

File: ompi/attribute/attribute.h (hunk 1 above)
Comment (hjelmn) [reply ompi/attribute/attribute.h:139]: Same as below. Aligning comments. We already do this in most of the code bar.

File: ompi/attribute/attribute_predefined.c (hunk 3 above)
Comment (hppritcha) [reply ompi/attribute/attribute_predefined.c:175]: yes this looks messy.  will remove the comments as part of this PR.

Comment (awlauria) [reply ompi/attribute/attribute_predefined.c:175]: I think it's fine. Comment cleanup can be done in another PR.

File: ompi/class/ompi_seq_tracker.c [hunk 5]
```diff
@@ -21,178 +21,163 @@
...
-                   NULL);
-
-
-static void ompi_seq_tracker_construct(ompi_seq_tracker_t* seq_tracker) {
+static void ompi_seq_tracker_construct(ompi_seq_tracker_t *seq_tracker)
+{
     OBJ_CONSTRUCT(&seq_tracker->seq_ids, opal_list_t);
-    seq_tracker->seq_ids_current = (ompi_seq_tracker_range_t*)opal_list_get_end(&seq_tracker->seq_ids);
+    seq_tracker->seq_ids_current = (ompi_seq_tracker_range_t *) opal_list_get_end(
+        &seq_tracker->seq_ids);
```
Comment (hppritcha) [inline ompi/class/ompi_seq_tracker.c:30]: this is messy line break

File: ompi/class/ompi_seq_tracker.c [hunk 6]
```diff
@@ -21,178 +21,163 @@
...
+    while (NULL != (item = opal_list_remove_first(&seq_tracker->seq_ids)))
         OBJ_RELEASE(item);
     OBJ_DESTRUCT(&seq_tracker->seq_ids);
 }
 
-OBJ_CLASS_INSTANCE(
-                   ompi_seq_tracker_t,
-                   opal_object_t,
-                   ompi_seq_tracker_construct,
+OBJ_CLASS_INSTANCE(ompi_seq_tracker_t, opal_object_t, ompi_seq_tracker_construct,
```
Comment (hppritcha) [inline ompi/class/ompi_seq_tracker.c:41]: this reorder makes the code less clear

**Comment by awlauria** [comment]
I checked the copyrights on this branch, and they do not look munged to my eye. I vote we do not go out in the weeds on any other changes but formatting and just get it in. The other changes needed (header ordering/comments) can come in separately.

File: ompi/class/ompi_seq_tracker.c [hunk 7]
```diff
@@ -21,178 +21,163 @@
...
+        } else if (seq_id < item->seq_id_low) {
+            if (direction == 1) {
                 /* we have gone forward in the list, and we went one item too far */
                 new_item = OBJ_NEW(ompi_seq_tracker_range_t);
                 new_item->seq_id_low = new_item->seq_id_high = seq_id;
-                opal_list_insert_pos(seq_ids,
-                                     (opal_list_item_t*) item,
-                                     (opal_list_item_t*) new_item);
+                opal_list_insert_pos(seq_ids, (opal_list_item_t *) item,
+                                     (opal_list_item_t *) new_item);
```
Comment (hppritcha) [inline ompi/class/ompi_seq_tracker.c:155]: code is less readable after reformatting

File: ompi/class/ompi_seq_tracker.h [hunk 8]
```diff
@@ -34,37 +34,33 @@ typedef struct ompi_seq_tracker_range_t ompi_seq_tracker_range_t;
...
 OMPI_DECLSPEC OBJ_CLASS_DECLARATION(ompi_seq_tracker_range_t);
 
-struct ompi_seq_tracker_t{
-    opal_list_t seq_ids;         /**< list of seqs id's that have been seen */
-    ompi_seq_tracker_range_t* seq_ids_current; /**< a pointer to the last place we were in the list */
-
+struct ompi_seq_tracker_t {
+    opal_list_t seq_ids; /**< list of seqs id's that have been seen */
+    ompi_seq_tracker_range_t
+        *seq_ids_current; /**< a pointer to the last place we were in the list */
```
Comment (hppritcha) [inline ompi/class/ompi_seq_tracker.h:40]: hopefully fixable by splitting comment across two lines

File: ompi/group/group_bitmap.c [hunk 9]
```diff
@@ -22,56 +22,55 @@
...
             ranks2[j] = MPI_UNDEFINED;
             m = ranks1[j];
             count = 0;
-            tmp = ( 1 << (m % BSIZE) );
+            tmp = (1 << (m % BSIZE));
             /* check if the bit that correponds to the parent rank is set in the bitmap */
-            if ( tmp == (child_group->sparse_data.grp_bitmap.grp_bitmap_array[(int)(m/BSIZE)]
-                         & (1 << (m % BSIZE)))) {
+            if (tmp
+                == (child_group->sparse_data.grp_bitmap.grp_bitmap_array[(int) (m / BSIZE)]
```
Comment (hppritcha) [inline ompi/group/group_bitmap.c:56]: messy line break

File: ompi/group/group_bitmap.c [hunk 10]
```diff
@@ -22,56 +22,55 @@
...
-                for (i=0 ; i<=(int)(m/BSIZE) ; i++) {
-                    for (k=0 ; k<BSIZE ; k++) {
-                        tmp1 = ( 1 << k);
-                        if ( tmp1 == ( child_group->sparse_data.grp_bitmap.grp_bitmap_array[i]
-                                       & (1 << k) ) ) {
+                for (i = 0; i <= (int) (m / BSIZE); i++) {
+                    for (k = 0; k < BSIZE; k++) {
+                        tmp1 = (1 << k);
+                        if (tmp1
+                            == (child_group->sparse_data.grp_bitmap.grp_bitmap_array[i]
```
Comment (hppritcha) [inline ompi/group/group_bitmap.c:67]: another messy line break

File: ompi/group/group_bitmap.c [hunk 11]
```diff
@@ -82,33 +81,31 @@ int ompi_group_translate_ranks_bmap ( ompi_group_t *parent_group,
...
-            for (i=0 ; i<child_group->sparse_data.grp_bitmap.grp_bitmap_array_len ; i++) {
-                for (k=0 ; k<BSIZE ; k++) {
-                    tmp = ( 1 << k);
-                    if ( tmp == ( child_group->sparse_data.grp_bitmap.grp_bitmap_array[i]
-                                  & (1 << k) ) ) {
+            for (i = 0; i < child_group->sparse_data.grp_bitmap.grp_bitmap_array_len; i++) {
+                for (k = 0; k < BSIZE; k++) {
+                    tmp = (1 << k);
+                    if (tmp
+                        == (child_group->sparse_data.grp_bitmap.grp_bitmap_array[i] & (1 << k))) {
```
Comment (hppritcha) [inline ompi/group/group_bitmap.c:104]: another messy line break

File: ompi/group/group_init.c [hunk 12]
```diff
@@ -141,14 +138,14 @@ ompi_group_t *ompi_group_allocate_sporadic(int group_size)
     }
     /* allocate array of (grp_sporadic_list )'s */
     if (0 < group_size) {
-        new_group->sparse_data.grp_sporadic.grp_sporadic_list =
-            (struct ompi_group_sporadic_list_t *)malloc
-            (sizeof(struct ompi_group_sporadic_list_t ) * group_size);
+        new_group->sparse_data.grp_sporadic.grp_sporadic_list
+            = (struct ompi_group_sporadic_list_t *) malloc(sizeof(struct ompi_group_sporadic_list_t)
+                                                           * group_size);
```
Comment (hppritcha) [inline ompi/group/group_init.c:143]: this is messy

File: ompi/group/group_init.c [hunk 13]
```diff
@@ -209,20 +206,20 @@ ompi_group_t *ompi_group_allocate_bmap(int orig_group_size , int group_size)
...
     /* allocate the unsigned char list */
-    new_group->sparse_data.grp_bitmap.grp_bitmap_array = (unsigned char *)malloc
-        (sizeof(unsigned char) * ompi_group_div_ceil(orig_group_size,BSIZE));
+    new_group->sparse_data.grp_bitmap.grp_bitmap_array = (unsigned char *) malloc(
+        sizeof(unsigned char) * ompi_group_div_ceil(orig_group_size, BSIZE));
 
-    new_group->sparse_data.grp_bitmap.grp_bitmap_array_len =
-        ompi_group_div_ceil(orig_group_size,BSIZE);
+    new_group->sparse_data.grp_bitmap.grp_bitmap_array_len = ompi_group_div_ceil(orig_group_size,
+                                                                                 BSIZE);
```
Comment (hppritcha) [inline ompi/group/group_init.c:213]: is there  a way to get clang-format to not change these type of assignment operations?

**Review by hppritcha — COMMENTED** [review]
certainly more things to find but this is enough for now.

**Comment by hppritcha** [comment]
> I checked the copyrights on this branch, and they do not look munged to my eye. I vote we do not go out in the weeds on any other changes but formatting and just get it in. The other changes needed (header ordering/comments) can come in separately.

I agree.  We can leave additional formating changes to subsequent PRs.

**Comment by hjelmn** [comment]
Yup. The biggest issue with clang-format seems to be that even though it should prefer to break on assignment it breaks before the first argument of the function. These are usually best to rework on a case-by-case basis as needed. For now  this looks ok to me.

**Comment by awlauria** [comment]
@hppritcha is this ready to be switched from a draft?

**Comment by hppritcha** [comment]
@awlauria I'm fine with merging this.  Might as well get this over with.

**Comment by awlauria** [comment]
I'm fine with it as well.

**Review by awlauria — DISMISSED** [review]

**Comment by hppritcha** [comment]
I found a handy git clang-format integration tool of unknown provenance, but it seems to work at least on staged files.
  
https://gist.github.com/spott/98e25623ebc1d923dce4

**Comment by gpaulsen** [comment]
@hppritcha That gist you found, appears to be an early version of what's included in the llvm project here:
https://github.com/llvm-mirror/clang/blob/master/tools/clang-format/git-clang-format

**Review by devreal — COMMENTED** [review]
I really don't have time to go through all >100k lines changed (I doubt anyone has) so I randomly picked some files, plus some of the code I care about. Too many unnecessary whitespace changes, most of which don't improve anything (granted, there are some egregious issues that are being fixed but those don't justify 100k patch). I've pointed out a couple of changes where readability is severely impaired (for me at least, code formatting is a highly subjective matter after all). Abdicating human control over formatting is a mistake, IMHO, and I can only shake my head about the suggestion that we are now coding against clang-format and will spend more time fixing formatting this PR breaks.

I understand the argument that we should merge this PR because we have merged the opal reformatting. If consistency is the issue, I'd volunteer to revert the opal changes (there are 84 conflicts, manageable and probably easy to fix).

**Comment by bosilca** [comment]
I have to agree with @devreal here, many of these changes look terrible, and fail to either provide a semantic change or improve the readability of the code. Zero benefit but significant drawback because as many of the contributors are familiar with the code they contributed, not only because of the logic constructs, but also because of the shape and organization. We are losing all this with this series of commits, putting additional, unnecessary, burden on contributors. I remain supportive of setting some basic code consistency rules, and encouraging developers to follow them. All this is good principle. But policing others developers code, it's not smart strategy.

**Comment by manjugv** [comment]
@hppritcha bought this is to my attention. Thanks!. @yosefe @jladd-mlnx @brminich You might want to review and provide your opinion here.

File: ompi/mca/osc/ucx/osc_ucx_passive_target.c [hunk 14]
```diff
@@ -278,12 +285,13 @@ int ompi_osc_ucx_flush(int target, struct ompi_win_t *win) {
...
-    ompi_osc_ucx_module_t *module = (ompi_osc_ucx_module_t *)win->w_osc_module;
+int ompi_osc_ucx_flush_all(struct ompi_win_t *win)
+{
+    ompi_osc_ucx_module_t *module = (ompi_osc_ucx_module_t *) win->w_osc_module;
     int ret = OMPI_SUCCESS;
 
-    if (module->epoch_type.access != PASSIVE_EPOCH &&
-        module->epoch_type.access != PASSIVE_ALL_EPOCH) {
+    if (module->epoch_type.access != PASSIVE_EPOCH
+        && module->epoch_type.access != PASSIVE_ALL_EPOCH) {
```
[the remark this answers is not in the document]
Comment (hjelmn) [reply ompi/mca/osc/ucx/osc_ucx_passive_target.c:294]: Have to disagree :). What is aligned here can be tweaked (operator vs plans) but i find putting the operator at the beginning of the line to be gay more readable.

**Comment by hjelmn** [comment]
> I really don't have time to go through all >100k lines changed (I doubt anyone has) so I randomly picked some files, plus some of the code I care about. Too many unnecessary whitespace changes, most of which don't improve anything (granted, there are some egregious issues that are being fixed but those don't justify 100k patch). I've pointed out a couple of changes where readability is severely impaired (for me at least, code formatting is a highly subjective matter after all). Abdicating human control over formatting is a mistake, IMHO, and I can only shake my head about the suggestion that we are now coding against clang-format and will spend more time fixing formatting this PR breaks.
> 
> I understand the argument that we should merge this PR because we have merged the opal reformatting. If consistency is the issue, I'd volunteer to revert the opal changes (there are 84 conflicts, manageable and probably easy to fix).

Can't say I agree. I have spent quite a bit of time with style consistency with changes contributed to code I owned. Leaving it to a machine will stop that. There are minor issues here and there that can be fixed easily. New code will have to follow the formatting guidelines and it is relatively easy to make of formatting readable (the vast minority).

File: ompi/mca/osc/ucx/osc_ucx_passive_target.c (hunk 14 above)
[the remark this answers is not in the document]
Comment (jsquyres) [reply ompi/mca/osc/ucx/osc_ucx_passive_target.c:294]: This is clearly a difference of stylistic preference -- there is no right answer here.  Just to make that point: I definitely have a preference on this matter, too, but I'm going to refrain from making my opinion known here.

This is exactly why the original coding standards were "light touch": we knew that we couldn't / wouldn't agree on lots of style preferences, so we only mandated a small set that we all agreed on.

Putting `&&` or `||` at the beginning or ends of lines falls into the "we're not going to agree" category.  It should not be mandated one way or the other via clang-format.

Can we disable this issue in the config file?

[the remark this answers is not in the document]
Comment (yosefe) [reply ompi/mca/osc/ucx/osc_ucx_passive_target.c:294]: 2 cents from clang-format usage in ucx: You must make a decision. clang-format basically tokenizes the code and then re-generates the text according to the defined style. It's not like checkpatch.

**Comment by yosefe** [comment]
In UCX we decided to require only new added code to follow the strict clang-format style, and keep existing code as-is to avoid large changes. This is done by running git clang-format tool in CI.

**Comment by hjelmn** [comment]
> In UCX we decided to require only new added code to follow the strict clang-format style, and keep existing code as-is to avoid large changes. This is done by running git clang-format tool in CI.

I, personally, like this approach. It allows the code author to catch and fix any weird formatting in the code when they change it.

**Comment by rhc54** [comment]
My own $0.02...

We walked away from this in the PMIx and PRRTE repos. It felt like we were taking a howitzer to what was really a rather minor matter. The number of places where the code didn't meet the "light touch" rules was rather small, and nobody was dying from format violations. We tried clang-format per the OMPI support rules/scripts, and it created a giant mess of weird line breaks and hard-to-read tables.

So we've gone back to simply "fixing" things on an as-we-go basis, educating developers with comments when needed. Ultimately, I think that the people who write and maintain the code should (for the most part) do so in a manner they best understand. The only "line in the sand" I draw is that the code be organized - there shouldn't be multiple format styles in the same function, for example.

Anyway, OMPI can do what it wants in this regard. Just offering my thoughts.

**Comment by bosilca** [comment]
I think people's code has more information that the code and comments themselves, information not meant for computers but for other humans reading the code. How many times when reviewing a code, even if it does not follow the exact same formatting principles, after a few hundreds of lines, patterns become apparent, logical grouping, that simplify the reading. All this additional information will be lost, in a strict, column restricted, syntax-free machine driven reformatting. I would agree this is a subjective view, but I can also make an objective argument. Collectively we pointed out quite a few places where the resulting code is difficult to read, enough in my view to relax the way we apply this reformatting.

**Comment by hppritcha** [comment]
closing.  need a different tool that generates fewer problems.

**Pull request closed without merging**